src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -448,20 +475,24 @@
         this)
       (a/stop [this]
         (shutdown! app-context)
-        this))))
+        this)
+      (a/restart [this]
+        (run-lifecycle-fns app-context s/stop "stop" (reverse ordered-services))
```

Comment:
I think in `shutdown` we'll catch an Exception thrown from an individual service's stop function, log it, and continue calling the other services so they still have a chance to stop?  Should we do that here too?  Maybe we should share some of this logic between the restart and `shutdown!`?